ziti/edge/types.go

```diff
@@ -170,3 +181,188 @@ func ApiResponseDecode(data interface{}, resp io.Reader) (*ApiResponseMetadata,
 
 	return apiR.Meta, nil
 }
+
+type InteceptDialOptions struct {
+	ConnectTimeoutSeconds *int
+	Identity              *string
+}
+
+type PortRange struct {
+	Low  uint16
+	High uint16
+}
+
+type InterceptV1Config struct {
+	Addresses   []ZitiAddress
+	PortRanges  []*PortRange
+	Protocols   []string
+	SourceIp    *string
+	DialOptions *InteceptDialOptions `json:"dialOptions"`
+	Service     *Service
+}
+
+type DomainName string
+
+func (dn DomainName) Match(hostname string) int {
+	if len(dn) < 3 {
+		return -1
+	}
+
+	if dn[0] == '*' {
+		domain := string([]byte(dn)[1:])
+		if strings.HasSuffix(hostname, domain) {
+			return len(hostname) - len(domain)
+		} else {
+			return -1
+		}
+	} else {
+		if hostname == string(dn) {
+			return 0
+		} else {
+			return -1
+		}
+	}
+}
+
+func (pr *PortRange) Match(port uint16) int {
+	if pr.Low <= port && port <= pr.High {
+		return int(pr.High - pr.Low)
+	}
+	return -1
+}
+
+type ZitiAddress struct {
+	cidr   *net.IPNet
+	ip     net.IP
+	domain DomainName
+}
+
+func (self *ZitiAddress) Matches(v any) int {
+	if ip, ok := v.(net.IP); ok {
+		if self.ip != nil {
+			if ip.Equal(self.ip) {
+				return 0
+			} else {
+				return -1
+			}
+		}
+
+		if self.cidr != nil {
+			if self.cidr.Contains(ip) {
+				ones, bits := self.cidr.Mask.Size()
+				return bits - ones
+			} else {
+				return -1
+			}
+		}
+	} else if hostname, ok := v.(string); ok {
+		return self.domain.Match(strings.ToLower(hostname))
+	}
+
+	return -1
+}
+
+func (self *ZitiAddress) UnmarshalText(data []byte) error {
+	v := string(data)
+	if _, cidr, err := net.ParseCIDR(v); err == nil {
+		self.cidr = cidr
+		return nil
+	}
+
+	if ip := net.ParseIP(v); ip != nil {
+		self.ip = ip
+		return nil
+	}
+
+	if len(v) < 3 {
```

Comment:
is 3 the magic number of "{character}.{character}" is that what the three represents?  I expect that's what's happening? A comment would be helpful here explaining that magic number